PURL-TYPES.rst

```diff
@@ -73,6 +73,20 @@ bitbucket
 
       pkg:bitbucket/birkenfeld/pygments-main@244fd47e07d1014f0aed9c
 
+bitnami
+-------
+``bitnami`` for Bitnami-based packages:
+
+- The default repository is ``https://downloads.bitnami.com/files/stacksmith``.
```

Comment:
Is this accurate? I just requested a download for Wordpress and the link was `https://bitnami.com/redirect/to/2333652/bitnami-wordpress-6.2.2-r1-debian-11-amd64.ova` which was a 302 pointing to `https://downloads.bitnami.com/files/stacks/wordpress/6.2.2/bitnami-wordpress-6.2.2-r1-debian-11-amd64.ova`.

Perhaps the default repo should be `https://downloads.bitnami.com/files` or `https://downloads.bitnami.com/files/stacks`, but I did not see any `stacksmith` when downloading this package.

Reply:
The package is downloaded from `https://downloads.bitnami.com/files/stacksmith/wordpress-6.2.2-15-linux-${OS_ARCH}-debian-11.tar.gz` where could be `arm64` or `amd64` resulting in
- `https://downloads.bitnami.com/files/stacksmith/wordpress-6.2.2-15-linux-amd64-debian-11.tar.gz`
- `https://downloads.bitnami.com/files/stacksmith/wordpress-6.2.2-15-linux-amd64-debian-11.tar.gz`

The logic in charge of downloading the Bitnami packages in order to build the different container images is present in the Dockerfiles, for instance at https://github.com/bitnami/containers/blob/cb05fd7362822459917ce7b00dc49535a5810b30/bitnami/wordpress/6/debian-11/Dockerfile#L37

Reply:
That may be true for containers, but I retrieved a VirtualBox OVA which did not include `stacksmith` in the URL. See https://bitnami.com/stack/wordpress/virtual-machine

Reply:
The VirtualBox OVA is downloaded from `https://downloads.bitnami.com` but that's only the final `.ova` file you can use on your computer. Internally, it includes more than one Bitnami package: WordPress, Apache, PHP, MariaDB. Each of them located under the same `https://downloads.bitnami.com/files/stacksmith` domain:

* MariaDB: `https://downloads.bitnami.com/files/stacksmith/mariadb-10.11.4-0-linux-${OS_ARCH}-debian-11.tar.gz`
* PHP: `https://downloads.bitnami.com/files/stacksmith/php-8.2.7-3-linux-${OS_ARCH}-debian-11.tar.gz`
* Apache: `https://downloads.bitnami.com/files/stacksmith/apache-2.4.57-8-linux-${OS_ARCH}-debian-11.tar.gz`

And the purl pointing to those packages is available into the final image (for example: MariaDB file is at `/opt/bitnami/mariadb/.spdx-mariadb.spdx`)

```
bitnami@debian:~$ grep -C 3  "pkg:bitnami" /opt/bitnami/mariadb/.spdx-mariadb.spdx
                {
                    "referenceCategory": "PACKAGE-MANAGER",
                    "referenceType": "purl",
                    "referenceLocator": "pkg:bitnami/mariadb@10.11.4"
                }
            ]
```

Reply:
Thanks for the clarification. This is not obvious, but makes sense.